raiden-network/light-client
# Cooperative settle

Fixes #2839 #2850 
Uses contract from https://github.com/raiden-network/raiden-contracts/pull/1474

**Short description**
Cooperative settle. Requires raiden-contracts 0.39, but also works in a backwards compatible manner with current 0.37 (needs more testing).

**Definition of Done**


**Steps to manually test the change (dApp)**

1. Ensure git submodules, dependencies and builds are up to date with this PR
2. Run cli with the compatible PFS: `./raiden --pathfinding-service-address=https://pfs-goerli-with-fee.services-dev.raiden.network`
3. Open a channel between 2 CLIs on these contracts: `http -v PUT http://localhost:5001/api/v1/channels token_address=0xC563388e2e2fdD422166eD5E76971D11eD37A466 partner_address=<addr2> settle_timeout:=60`
4. Check the channels: `http http://localhost:5001/api/v1/channels`
4. Close the channel: `http -v PATCH http://localhost:5001/api/v1/channels/0xC563388e2e2fdD422166eD5E76971D11eD37A466/<addr2> state=closed`
5. Check the channels again and see it was **settled** directly.
